src/sentry/api/endpoints/project_stacktrace_link.py

```diff
@@ -155,17 +161,106 @@ def try_path_munging(
     return result, current_iteration_count
 
 
-def set_tags(scope: Scope, result: JSONData) -> None:
-    scope.set_tag("stacktrace_link.found", result["sourceUrl"] is not None)
-    scope.set_tag("stacktrace_link.source_url", result.get("sourceUrl"))
+def set_tags(scope: Scope, result: StacktraceLinkOutcome, integrations: List[None]) -> None:
+    scope.set_tag("stacktrace_link.found", result["source_url"] is not None)
+    scope.set_tag("stacktrace_link.source_url", result.get("source_url"))
     scope.set_tag("stacktrace_link.error", result.get("error"))
     scope.set_tag("stacktrace_link.tried_url", result.get("attemptedUrl"))
-    if result["config"]:
-        scope.set_tag("stacktrace_link.empty_root", result["config"]["stackRoot"] == "")
+    if result["current_config"]:
+        scope.set_tag(
+            "stacktrace_link.empty_root",
+            result["current_config"]["config"].automatically_generated == "",
+        )
         scope.set_tag(
-            "stacktrace_link.auto_derived", result["config"]["automaticallyGenerated"] is True
+            "stacktrace_link.auto_derived",
+            result["current_config"]["config"].automatically_generated is True,
         )
-    scope.set_tag("stacktrace_link.has_integration", len(result["integrations"]) > 0)
+    scope.set_tag("stacktrace_link.has_integration", len(integrations) > 0)
+
+
+class StacktraceLinkConfig(TypedDict):
+    config: RepositoryProjectPathConfig
+    outcome: ReposityLinkOutcome
+    repository: str
+
+
+class StacktraceLinkOutcome(TypedDict):
+    source_url: str | None
+    error: str | None
+    current_config: StacktraceLinkConfig | None
+    iteration_count: int
+    is_munged: bool
+
+
+def get_stacktrace_config(
```

Comment:
If we're sharing this code it might be better to figure out somewhere other than the endpoint for it